### PR TITLE
Introduce inverse of MF::get_cell_iterator()

### PR DESCRIPTION
**Pull request by peterrum**
@kronbichler As discussed. I am not sure about the names and also I was not sure if we should return `unsigned int` or a pair of `unsigned int`s.  In my codes, I normally found it easier to handle a simple `unsinged int` and translate it at relevant places (this is also what we do for the `FEEvaluation::reinit()` version that takes user defined macro cells).

### CODE REVIEW

File: include/deal.II/matrix_free/matrix_free.h [hunk 1]
```diff
@@ -1656,6 +1656,16 @@ class MatrixFree : public Subscriptor
   get_cell_level_and_index(const unsigned int cell_batch_index,
                            const unsigned int lane_index) const;
 
+  /**
+   * Get MatrixFree index associated to a deal.II @p cell. To get
+   * the actual cell batch index and land do the postprocessing
```
Comment (drwells) [inline include/deal.II/matrix_free/matrix_free.h:1661]:
```suggestion
   * the actual cell batch index and lane, do the postprocessing
```

**Review by drwells — APPROVED** [review]
Could we add a basic test?

**Comment by kronbichler** [comment]
I agree with the proposed name. I would have intuitively said we should return the `cell` and `lane` index, but since we are exposing the underlying index already, I think we could go with the easier design as done here.

Regarding this data structure, I believe we should use this in some of the setup function, e.g., here https://github.com/dealii/dealii/blob/master/include/deal.II/matrix_free/face_setup_internal.h#L747-L748 as well. This code was clearly written before we had the active cell index (even though committed to the main branch only in 2018, I believe this is code written in 2013) because `std::map` is a bad data structure for this kind of thing.

**Comment by peterrum** [comment]
I have moved the initialization and have added a test.

**Review by kronbichler — APPROVED** [review]

File: include/deal.II/matrix_free/matrix_free.templates.h [hunk 2]
```diff
@@ -1752,6 +1765,29 @@ namespace internal
...
+           cell < n_cell_batches() + n_ghost_cell_batches();
+           ++cell)
+        for (unsigned int v = 0; v < n_active_entries_per_cell_batch(cell); ++v)
+          {
+            const auto tria_cell = get_cell_iterator(cell, v);
+            mf_cell_indices[(mg_level == numbers::invalid_unsigned_int) ?
+                              tria_cell->active_cell_index() :
+                              tria_cell->index()] =
+              cell * VectorizedArrayType::size() + v;
+          }
```
Comment (kronbichler) [inline include/deal.II/matrix_free/matrix_free.templates.h:1788]: This won't compile, you need to be in a member function of `MatrixFree` to be able to use these variables. (See also the CI failures.)

Comment (peterrum) [reply include/deal.II/matrix_free/matrix_free.templates.h]: @kronbichler I have the code back to the old place. Let's us the information later...

**Review by kronbichler — APPROVED** [review]

**Pull request merged**